File: DI/DI-Dagger-Java/src/main/java/DI/persona/controllers/PersonasFileController.java
package DI.persona.controllers;

import DI.persona.modules.PersonasFileModule;
import DI.persona.modules.PersonasRepositoryModule;
import dagger.Component;
import model.persona.controllers.PersonaController;

import javax.inject.Singleton;

@Singleton
@Component(modules = {
        PersonasFileModule.class,
        PersonasRepositoryModule.class
})
public interface PersonasFileController {
    PersonaController build();
}

File: DI/DI-Dagger-Kotlin/src/main/kotlin/DI/personas/controllers/PersonasFileController.kt
package DI.personas.controllers

import DI.personas.modules.PersonasFileModule
import DI.personas.modules.PersonasRepositoryModule
import dagger.Component
import models.personas.controllers.PersonaController

import javax.inject.Singleton


@Singleton
@Component(
    modules = [
        PersonasFileModule::class,
        PersonasRepositoryModule::class
    ]
)
interface PersonasFileController {
    fun build(): PersonaController
}
